src/main/java/eu/interop/federationgateway/service/DiagnosisKeyBatchService.java

```diff
@@ -65,158 +61,140 @@
-          return;
-        }
-
-        diagnosisKeyCollector.addAll(diagnosisKeyEntitys);
-
-        if (checkBatchSize(diagnosisKeyCollector)) {
-          break;
-        }
-
-        // update keys
-        updateDiagnosisKeys(diagnosisKeyEntitys, newBatch);
-      }
-      MDC.put("batchCount", String.valueOf(diagnosisKeyCollector.size()));
-      log.info("Batch process finished");
-      callbackService.notifyAllCountriesForNewBatchTag(newBatch);
+    boolean batchCreationResult;
+    do {
+      batchCreationResult = createNextBatch();
 
       long elapsedTime = System.currentTimeMillis() - startTime;
       if (elapsedTime > properties.getBatching().getTimelimit()) {
         MDC.put("batchTime", String.valueOf(elapsedTime));
         log.info("Maximum time for one batching execution reached.");
         break;
       }
-    }
-  }
 
-  @Recover
-  public String recover(Throwable t) {
-    log.info("DiagnosisKeyBatchService.recover");
-    return "error class :: " + t.getClass().getName();
-  }
+      batchCount += batchCreationResult ? 1 : 0;
+    } while (batchCreationResult);
 
-  private List<DiagnosisKeyEntity> findAllKeysByBatchTag(String uploaderBatchTag) {
-    // find all keys by the uploader batch tag
-    List<DiagnosisKeyEntity> diagnosisKeyEntitys
-      = diagnosisKeyEntityRepository.findByBatchTagIsNullAndUploader_BatchTag(uploaderBatchTag);
-    log.info("found {} diagnosis keys with uploader tag {} for batching process",
-      diagnosisKeyEntitys.size(), uploaderBatchTag);
-    return diagnosisKeyEntitys;
-  }
+    MDC.put("batchCount", String.valueOf(batchCount));
+    log.info("Batch Process finished");
 
-  private boolean checkBatchSize(List<DiagnosisKeyEntity> diagnosisKeyCollector) {
-    if (diagnosisKeyCollector.size() > properties.getBatching().getDoclimit()) {
-      log.info("Continue the batching process to next batch while {} "
-          + "keys collected and the limit: {} of keys is reached",
-        diagnosisKeyCollector.size(), properties.getBatching().getDoclimit());
-      return true;
-    }
-    return false;
+
+    MDC.remove("batchCount");
+    MDC.remove("batchTime");
   }
 
-  private boolean checkUploaderLimitation(List<DiagnosisKeyEntity> diagnosisKeyEntitys, String uploaderBatchTag) {
-    if (diagnosisKeyEntitys.size() > properties.getBatching().getDoclimit()) {
-      log.error("Stop batching process, while try to batch {} keys from one uploader: {}, "
-          + "but the configured batching limit of keys is {}",
-        diagnosisKeyEntitys.size(), uploaderBatchTag, properties.getBatching().getDoclimit());
-      return true;
+  /**
+   * Creates a new Batch Entity and sets batchTag to all contained diagnosiskeys.
+   *
+   * @return true if a batch was created or false if not.
+   */
+  @Transactional(Transactional.TxType.REQUIRES_NEW)
+  public boolean createNextBatch() {
+
+    List<String> uploaderBatchTags = collectUploaderBatchTags();
+    log.info("collected batch tags");
```

Comment:
possibly remove this message or add values to it and then add it to the documentation

Reply:
Forgot to remove it